spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java

```diff
@@ -86,7 +86,12 @@
 	 * Use ip address rather than hostname during registration
 	 */
 	private boolean preferIpAddress = false;
-
+	
+	/**
+	 * Source of how we will determine the address to use
+	 */
+	private AddressSource addressSource = AddressSource.AGENT;
```

Comment:
Yea enum probably isn't but did it that way for possibly future enhancement to support possibly named interface.  I can remove that if needed, figured an enum is pretty lightweight and simple.